python/torch_mlir/csrc/backend/mlir_node_lowering.h

```diff
@@ -0,0 +1,30 @@
+//===- mlir_node_lowering.h -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+// This file is adapted from pytorch/pytorch
+// https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/ts_backend/ts_node_lowering.h
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <torch/csrc/api/include/torch/jit.h>
+#include <torch/csrc/lazy/backend/lowering_context.h>
+
+namespace torch {
+namespace lazy {
+
+typedef std::vector<torch::jit::Value*> TorchMlirOpVector;
+typedef std::shared_ptr<torch::jit::GraphFunction> TorchMlirFunction;
+
+TORCH_API TorchMlirOpVector LowerTorchMlirBuiltin(
```

Comment:
Hm I think it would make sense to go with the LLVM coding style, since this part is under the LLVM org? Were there any parts you were particular concerned about? (e.g. variable naming convention)

Reply:
yeah, I added a `.clang-format` to the `csrc` directory that we can use for formatting. Its based off of the `LLVM` standard with a couple minor tweaks.

As for naming, you're right in that we've kept with the naming conventions of LTC in upstream PyTorch. Though, I agree that we should follow LLVM style guides as this is a project under LLVM. Converting the naming conventions to match LLVM conventions should be fairly trivial.

Reply:
The convention we have used in the past for the C++ code in csrc is to use PyTorch coding style, with `lowerCamelCase` local variables (PyTorch doesn't seem to be super consistent about the convention for local variable names, though looking at it now in retrospect `snake_case` seems predominant; class member names are also not super consistent across the PT codebase either).

My concrete concern is that a lot of the LTC code is in google style with `UpperCamelCase` function names. It would be great if upstream fixed that, since I don't think any other parts of the PyTorch codebase use that style and it is confusing.